### ISSUE TITLE
DefaultTypeHierarchy.visitTypevarSubtype can ignore type arguments

### ISSUE THREAD

**Issue by cpovirk** [typetools/checker-framework#3443]
```java
$ cat OnlyTopLevelForTypeVariable.java
import org.checkerframework.checker.nullness.qual.Nullable;

class OnlyTopLevelForTypeVariable {
  static <T extends Supplier<@Nullable String>> Supplier<String> passThrough(T t) {
    return t;
  }

  public static void main(String[] args) {
    Supplier<@Nullable String> s1 = () -> null;
    Supplier<String> s2 = passThrough(s1);
    s2.get().toString();
  }
}

interface Supplier<T> {
  T get();
}

$ $HOME/checker-framework-3.4.1/checker/bin/javac -processor org.checkerframework.checker.nullness.NullnessChecker OnlyTopLevelForTypeVariable.java && java OnlyTopLevelForTypeVariable
Exception in thread "main" java.lang.NullPointerException
        at OnlyTopLevelForTypeVariable.main(OnlyTopLevelForTypeVariable.java:11)
```

I would have expected an error in the implementation of `passThrough`, since a `Supplier<@Nullable String>` isn't a `Supplier<String>` (as the Checker Framework knows if I use `Supplier<@Nullable String>` directly instead of `T`).

I think this is a result of the code somewhere around here:
https://github.com/typetools/checker-framework/blob/4bd0591172fbf54cf128174589d68aebb2825eab/framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java#L1023

As I understand it, we end up checking `isPrimarySubtype` but not the arguments.

I didn't encounter this in "real" code.
